File: src/ExternalLibraries/Trading/Trading.Database/Interfaces/ITradingDatabaseHistoryInterface.cs
using Trading.Library.Classes;

namespace Trading.Database.Interfaces
{
    /// <summary>
    /// History of trading
    /// </summary>
    public interface ITradingDatabaseHistoryInterface
    {
        /// <summary>
        /// Deletes a symbol
        /// </summary>
        /// <param name="symbol"></param>
        Task DeleteBySymbolAsync(string symbol, CancellationToken token);

        void DeleteBySymbol(string symbol);

        /// <summary>
        /// Fill historty
        /// </summary>
        /// <param name="name">Name of symbol</param>
        /// <param name="data">Historical data</param>

        Task FillHisroryAsync(string name, List<HistoricalDataMessageDateTime> data, CancellationToken token);

        void FillHisrory(string name, List<HistoricalDataMessageDateTime> data);


        /// <summary>
        /// Gets historical message
        /// </summary>
        /// <param name="id">Id</param>
        /// <param name="begin">Begin</param>
        /// <param name="end">End</param>
        /// <returns>The messages</returns>
        Task<List<HistoricalDataMessageDateTime>> GetHistoricalDataMessageDateTimes(object id,
            DateTime begin, DateTime end, CancellationToken token);

        List<HistoricalDataMessageDateTime> GetHistoricalDataMessageDateTimes(object id,
            DateTime begin, DateTime end);




        /// <summary>
        /// Symbols
        /// </summary>
        Task<Dictionary<string, object>> GetSymbolsAsync(CancellationToken token);

      //  Dictionary<string, object> Symbols { get; }

    }
}

File: Java_Generated/Java/Maven/src/main/java/trading/database/interfaces/ITradingDatabaseHistoryInterface.java
package trading.database.interfaces;

import cancellation.interfaces.ICancellation;
import trading.database.classes.HistoricalDataMessageDateTime;

import java.util.Map;
import java.util.Objects;
import java.util.concurrent.CompletableFuture;

public interface ITradingDatabaseHistoryInterface {

    CompletableFuture<Map<String, Object>> getSymbolsAsync(ICancellation cancellation);
    CompletableFuture<HistoricalDataMessageDateTime[]> getHistoricalDataMessageDateTimesAsync(Object id, double begin, double end, ICancellation cancellation);

}
